File: padel-maniac-backend/src/main/java/org/example/padelmaniacbackend/service/MatchService.java
package org.example.padelmaniacbackend.service;

import org.example.padelmaniacbackend.DTOs.matchDTO.CreateMatchDTO;
import org.example.padelmaniacbackend.DTOs.matchDTO.MatchDTO;
import org.example.padelmaniacbackend.DTOs.matchDTO.MatchUnsubscribeOrJoinDTO;

import java.util.List;

public interface MatchService {

    public void createNewMatch(CreateMatchDTO createMatchDTO, String username);

    public List<MatchDTO> getMatches();

    public MatchDTO joinToMatch(MatchUnsubscribeOrJoinDTO matchUnsubscribeOrJoinDTO);

    public MatchDTO requestForMatch(MatchUnsubscribeOrJoinDTO matchUnsubscribeOrJoinDTO);

    public MatchDTO rejectRequest(MatchUnsubscribeOrJoinDTO matchUnsubscribeOrJoinDTO);

    public MatchDTO matchDetails(Long matchId);

    public MatchDTO removeMatch(Long matchId);

    public List<MatchDTO> getUpcomingMatches(Long playerId);

    public MatchDTO matchUnsubscribe(MatchUnsubscribeOrJoinDTO matchUnsubscribeOrJoinDTO);
}

File: padel-maniac-frontend/src/services/MatchService.js
import api from "./httpIntercepter";
import axios from "axios";

export default {

    async createMatch(formData){
        return api.post("/match/create", formData)
    },

    async getMatches(){
        return api.get("/match/getMatches")
    },

    async joinToMatch(matchId, playerId){
        return api.post("/match/joinToMatch",{
            matchId : matchId,
            playerId : playerId
        },{
            headers: {
                'Content-Type': 'application/json'
            }
        });
    },

    async requestForMatch(matchId,userId){
        return api.post("/match/requestForMatch",
            {
                matchId : matchId,
                playerId : userId
            }
            ,{
            headers: {
                'Content-Type': 'application/json'
            }
        });
    },

    async mathcDetails(matchId){
        return api.post("/match/matchDetails", matchId,{
            headers: {
                'Content-Type': 'application/json'
            }
        });
    },

    async removeMatch(matchId){
        return api.post("/match/removeMatch", {matchId : matchId},{
            headers: {
                'Content-Type': 'application/json'
            }
        });
    },

    async getUpcomingMatches(playerId){
        return api.post("/match/getUpcomingMatches",{playerId: playerId},{
            headers: {
                'Content-Type': 'application/json'
            }
        })
    },

    async matchUnsubscribe(matchId, playerId) {
        return api.post(
            "/match/matchUnsubscribe",
            {
                matchId: matchId,
                playerId: playerId
            }
        );
    },

    async rejectRequest(matchId, playerId) {
        return api.post(
            "/match/rejectRequest",
            {
                matchId: matchId,
                playerId: playerId
            }
        );
    }


}
